dexterion.cpp

```diff
@@ -28,11 +31,55 @@ LRESULT Wndproc(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam) {
 	return DefWindowProc(hWnd, msg, wParam, lParam);
 }
 
+// Function to get the PID of a process by its name
+DWORD GetPIDByName(const wchar_t* processName) {
+	DWORD pid = 0;
+	HANDLE hSnap = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
+	if (hSnap != INVALID_HANDLE_VALUE) {
+		PROCESSENTRY32W pe32; // Using PROCESSENTRY32W for wide strings (wchar_t)
+		pe32.dwSize = sizeof(PROCESSENTRY32W);
+		if (Process32FirstW(hSnap, &pe32)) { // Using Process32FirstW for wide strings (wchar_t)
+			do {
+				if (_wcsicmp(pe32.szExeFile, processName) == 0) { // Using _wcsicmp for wide strings (wchar_t) comparison
+					pid = pe32.th32ProcessID;
+					break;
+				}
+			} while (Process32NextW(hSnap, &pe32)); // Using Process32NextW for wide strings (wchar_t)
+		}
+		CloseHandle(hSnap);
+	}
+	return pid;
+}
 
 int APIENTRY WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nShowCmd) {
 	AllocConsole();
 	freopen("CONOUT$", "w", stdout);
 	HANDLE hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
+
+	// Execute updateoffset.cmd in order to update before run
+	SetConsoleTextAttribute(hConsole, 9);
+	printf("[MemMan] Updating required files\n");
+	SetConsoleTextAttribute(hConsole, 15);
+	system("cmd.exe /C updateoffset.cmd");
+	SetConsoleTextAttribute(hConsole, 10);
+	printf("[MemMan] Required files updated\n");
+
+	DWORD cs2Pid = 0;
+
+	while (cs2Pid == 0) {
+		// Search for cs2.exe among the PIDs
+		cs2Pid = GetPIDByName(L"cs2.exe");
+		if (cs2Pid != 0) {
+			SetConsoleTextAttribute(hConsole, 10);
+			std::wcout << L"cs2.exe found! PID: " << cs2Pid << std::endl;
+		}
+		else {
+			SetConsoleTextAttribute(hConsole, 12);
+			std::wcout << L"cs2.exe not found. Retrying in 5 seconds..." << std::endl;
+			Sleep(5000); // Wait for 5 seconds before the next attempt
+		}
+	}
+	
 	// Memory and game related vars (used in entry and passed through overlay)
 	int procId = MemMan.getPid(L"cs2.exe");
```

Comment:
Why are you still keeping this part of the code if you're already parsing the proccess id above?

Reply:
Because using cs2Pid don't work here 
https://github.com/Skwrr/Dexterion/blob/ff17b58ddedafa591088debb7d410c74d446af3e/dexterion.cpp#L67
Replacing procId

Reply:
Maybe I need to convert it into Int in order to work